File: src/api/AMP.EMS.API/Core/Entities/EventGuest.cs
using System;
using AMP.Infrastructure.Entity;

namespace AMP.EMS.API.Core.Entities;

public class EventGuest : FullAuditableEntity<Guid>
{
    public required Guid EventId { get; set; }
    public required Guid GuestId { get; set; }
    public required int MaxGuests { get; set; }
    public Event? Event { get; set; }
    public Guest? Guest { get; set; }
    public ICollection<EventGuestInvitation> EventGuestInvitations { get; set; } = [];
    public ICollection<EventGuestRole> EventGuestRoles { get; set; } = [];
}

File: src/ui/AMP.EMS/src/app/core/models/event-guest.ts
import { Guest } from "./guest";
import { Event } from "./event";
import { EventGuestInvitation } from "./event-guest-invitation";
import { EventGuestRole } from "./event-guest-role";

export interface EventGuest {
    id?: string;
    eventId?: string;
    guestId?: string;
    maxGuests?: number;
    guest?: Guest;
    eventGuestInvitations?: EventGuestInvitation[];
    eventGuestRoles?: EventGuestRole[];
}